pkg/controller/glustercluster/reconcile.go

```diff
@@ -0,0 +1,103 @@
+func (r *ReconcileGlusterCluster) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling GlusterCluster")
+
+	// Fetch the GlusterCluster instance
+	instance := &operatorv1alpha1.GlusterCluster{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Get current reconcile version from CR
+	version := instance.Status.ReconcileVersion
+	if version == nil {
+		// choose the highest compatible version
+		reconcileProcedure, err = allProcedures.Newest()
+		if err != nil {
+			log.Error(err, "Failed to get newest reconcile procedure")
+		}
+	} else {
+		// If no current version, use highest version to reconcile
+		reconcileProcedure, err = allProcedures.NewestCompatible(*version)
+		if err != nil {
+			log.Error(err, "Failed to get newest compatible reconcile procedure.")
+		}
+	}
+
+	// Execute the reconcile procedure. Not sure how to handle the error
+	procedureStatus, err = reconcileProcedure.Execute(request, r.client, r.scheme)
+	if err != nil {
+		log.Error(err, "Failed to execute procedure.")
+		return reconcile.Result{}, err
+	}
+	// Walk ProcedureStatus.Results and add to the CR status
+	reconcileActionStatus := make(map[string]reconciler.Result)
+	for _, result := range procedureStatus.Results {
+		reconcileActionStatus[result.Name] = result.Result
+	}
+	instance.Status.ReconcileActions = reconcileActionStatus
+
+	if !procedureStatus.FullyReconciled {
+		err = r.client.Update(context.TODO(), instance)
+		if err != nil {
+
+			return reconcile.Result{}, err
+		}
+		//requeue immediately
+		return reconcile.Result{Requeue: true}, nil
+	}
+	// if ProcedureStatus.FullyReconciled
+	//   update reconcile version in the CR to match the Procedure version
+	//   use a timed reconcile requeue //left this part out. Why requeue?
+	newVersion := reconcileProcedure.Version()
+	instance.Status.ReconcileVersion = &newVersion
+	err = r.client.Update(context.TODO(), instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	// use a timed reconcile requeue //TODO: implement backoff
+	return reconcile.Result{RequeueAfter: 3e+10}, nil
```

Comment:
How long a delay is this? I'd suggest 30-60 seconds.

Reply:
3e+10 nanoseconds = 30 seconds